src/RRTStar/RRTStarWrapper.cpp

```diff
@@ -0,0 +1,167 @@
+    //      yPath: list to store y path
+    int ApplyRRTStar(double xStart, double yStart, double xEnd, double yEnd,
+            int stepSize, int maxIterations, double* obstacles_llx,
+            double* obstacles_lly, double* obstacles_urx, double* obstacles_ury,
+            int numObstacles, double* xPath, double* yPath)
+    {
+        RRT* rrt = new RRT;
+
+        // Initialize configuration
+        rrt->reset();
+        rrt->setStateSpace(xStart, yStart, xEnd, yEnd);
+        rrt->setStepSize(stepSize);
+        rrt->setMaxIterations(maxIterations);
+        rrt->initialize();
+
+        // Construct obstacles
+        for (int i = 0; i < numObstacles; i++) {
+            rrt->addObstacle(
+                    Vector2f(obstacles_llx[i], obstacles_lly[i]),
+                    Vector2f(obstacles_urx[i], obstacles_ury[i])
+            );
+        }
+
+        // Declare variables
+        Node *qBest, *qNew, *qNearest;
+        double cost, dist, distBest, gamma, radius;
+        int reached = 0;
+
+        // Cost to each vertex
+        map<Node *, double> cost_map;
+        cost_map[rrt->root] = 0;
+
+        // Run RRT*
+        gamma = 1 + pow(2, SPACEDIM) * (1 + 1 / SPACEDIM) *
+                rrt->getFreeArea();
+        for (int i = 0; i < rrt->max_iter; i++) {
+            qNew = rrt->getRandomNode();
+            qNearest = rrt->nearest(qNew->position);
+            if (rrt->distance(qNew->position, qNearest->position) >
+                rrt->step_size) {
+                Vector2f newConfig = rrt->newConfig(qNew, qNearest);
+                if (rrt->isSegmentInObstacles(newConfig, qNearest->position)) {
+                    continue;
+                }
+                qNew = new Node;
+                qNew->position = newConfig;
+            }
+
+            rrt->nearestNeighbors(qNew->position);
+            qBest = qNearest;
+            distBest = rrt->distance(qNew->position, qNearest->position);
+            radius = min(
+                pow(gamma / PI * log(i + 1) / (i + 1), 1 / SPACEDIM),
+                (double) rrt->step_size
+            );
+
+            // look for shortest cost path to qNew
+            for (auto node : rrt->nodes) {
+                // compute the cost for qNew through node
+                dist = rrt->distance(qNew->position, node->position);
+                cost = cost_map[node] + dist;
+
+                // only consider proximal nodes
+                if (dist > radius) {
+                    break;
+                }
+
+                // if cost less than current lowest cost, check potential link
+                if (cost < cost_map[qBest] + distBest) {
+                    if (rrt->isSegmentInObstacles(qNew->position,
+                                                  node->position)) {
+                        continue;
+                    }
+                    qBest = node;
+                    distBest = dist;
+                }
+            }
```

Comment:
Can you please move this into a separate method? It will help to make the ApplyRRTStar method shorter/easier to understand.

Reply:
refactored!